STYLEGUIDE.adoc

```diff
@@ -0,0 +1,541 @@
+= Antaeus Haskell style guide
+:toc:
+
+The purpose of this document is to codify various aspects of how we write Haskell code as we contribute to Antaeus.
+
+The goal is twofold:
+
+1. To provide guidance in cases where it is not obvious what to do, or the choice is arbitrary.
+2. To help resolve disagreements between team members. Ideally these can be settled by referring to a recommendation in this document, or if not, then perhaps the resolution of the disagreement will create a new recommendation.
+
+The contents of this document are “recommendations”, that is:
+
+- They are more concrete than “principles” (like “write simple code”) because we want them to be clearly action-guiding.
+
+- They are more vague than “rules” (like “never use function X”) because it’s usually impossible to be that precise, and so judgement is still required in applying them.
+
+== Haskell
+
+=== Basic formatting
+
+Use 2 spaces for indentation. Full-indent for `where` is recommended.
```

Comment:
>Full-indent for `where` is recommended.

Does that mean 4 spaces for where and then 2 for the function indent
```
    where
      function
```
or 2 for each (totalling 4)
```
  where
    function
```
or 4 for each?
```
    where
        function
```

I tend to do 2 for each

Reply:
Yes, I guess some people like to use a one-space indent for `where` so they can use two spaces for the functions within the where-block.